skrub/tests/test_deduplicate.py

```diff
@@ -95,3 +98,67 @@ def test__create_spelling_correction(seed: int = 123) -> None:
             spelling_correction.values[clusters == n].astype("int")
             == counts[clusters == n].max()
         ).all()
+
+
+@skip_if_no_parallel
+def test_parallelism():
+    # Test that parallelism works
+    X = make_deduplication_data(examples=['black', 'white', 'red'], entries_per_example=[500, 500, 500], prob_mistake_per_letter=0.3)
+    y = deduplicate(X, n_jobs=None)
+    times = {}
+    jobs = [1, 4]
+
+    for n_jobs in jobs:
+        start = time.time()
+        y_parallel = deduplicate(X, n_jobs=n_jobs)
+        assert_array_equal(y, y_parallel)
+        end = time.time()
+        timer = end - start
+        times[n_jobs] = timer
+    print(times)
+    # Test if execution time is shorter:
+    for i in range(len(jobs)-1):
+        assert times[jobs[i+1]] < times[jobs[i]]
+
+    # Test with threading backend
+    with joblib.parallel_backend("threading"):
+        y_threading = deduplicate(X, n_jobs=1)
```

Comment:
Why not test the time with both backends? To simplify the code, it could probably be parametrized.

Reply:
After doing so, threading gives terrible performance (after discussing with @jeremiedbb, probably because of the GIL), so right now, we don't use threading, just process-based backends (and leave it default, loky).